core/src/main/scala/org/bykn/bosatsu/rankn/DefinedType.scala

```diff
@@ -23,15 +23,26 @@ case class DefinedType[+A](
    * A type with exactly one constructor is a struct
    */
   def isStruct: Boolean = constructors.lengthCompare(1) == 0
+
+  /**
+   * A newtype is just a wrapper for another type.
+   * It could be removed statically from the program
+   */
+  def isNewType: Boolean =
+    constructors match {
+      case (_, _ :: Nil, _) :: Nil => true
```

Comment:
I've seen pattern matches like `(_, _ :: Nil, _)` a few times in this code and it makes me wonder if `(Constructor, List[(Bindable, Type)], Type)` is a little too complicated of a tuple. That it might be a bit more self documenting if you had a case class with named methods.